### PR TITLE
Feat/recist table

### PR DESCRIPTION
**Pull request by OzgeYurtsever**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -61,6 +89,15 @@ class App extends Component {
       pid: null,
       closeAll: 0,
       projectAdded: 0,
+      showReport: false,
```
Comment (OzgeYurtsever) [inline src/App.js:92]: this is not used anymore, unnecessary state.

File: src/App.js [hunk 2]
```diff
@@ -725,7 +872,14 @@ class App extends Component {
...
       progressUpdated,
       treeExpand,
-      expandLevel,
+      showReportsMenu,
+      showReport,
+      showWarning,
+      showConfirmation,
+      title,
+      message,
+      reportType,
```
Comment (OzgeYurtsever) [inline src/App.js:881]: Checkout these variables to clean the code.

File: src/components/annotationsList/reducer.js [hunk 3]
```diff
@@ -514,6 +543,20 @@ const asyncReducer = (state = initialState, action) => {
             },
           },
         });
+
+      // case SELECT_REPORT:
```
Comment (OzgeYurtsever) [inline src/components/annotationsList/reducer.js:547]: Clean code this is not needed.

File: src/components/annotationsList/types.js [hunk 4]
```diff
@@ -46,6 +47,8 @@ export const GET_PROJECT_MAP = "epadjs/annotationList/GET_PROJECT_MAP";
 export const SET_SEG_LABEL_MAP_INDEX =
   "epadjs/annotationList/SET_SEG_LABEL_MAP_INDEX";
 export const GET_TEMPLATES = "epadjs/annotationList/GET_TEMPLATES";
+// export const SELECT_REPORT = "epadjs/annotationList/SELECT_REPORT";
```
Comment (OzgeYurtsever) [inline src/components/annotationsList/types.js:50]: Clean code here

File: src/components/common/SelectModalMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      >
+        {options}
+      </div>
+    </div>
+  );
+};
+
+export default SelectModalMenu;
+
+// setWrapperRef = node => {
```
Comment (OzgeYurtsever) [inline src/components/common/SelectModalMenu.jsx:50]: Clean code here.

File: src/components/searchView/Report.jsx [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+const style = { width: 'auto', minWidth: 300, maxHeight: 800, height: 'auto' };
+const Report = props => {
+  const [node, setNode] = useState(null);
+  const [data, setData] = useState([]);
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [selectedRow, setSelectedRow] = useState(null);
+  const [selectedCol, setSelectedCol] = useState(null);
+  const [filteredPatients, setFilteredPatients] = useState({});
+
+  // const filterSelectedPatients = () => {
```
Comment (OzgeYurtsever) [inline src/components/searchView/Report.jsx:39]: Clean code here

File: src/components/searchView/Report.jsx [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+            id,
+            patientID,
+            projectID,
+            report,
+            data,
+            numofHeaderCols,
+            hideCols,
+            loadFilter,
+            props.index,
+            // onClose, //js code doesn't know anything about this
```
Comment (OzgeYurtsever) [inline src/components/searchView/Report.jsx:132]: clean here

File: src/components/searchView/Report.jsx [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+    setShowConfirmModal(false);
+  };
+
+  const header =
+    props.report !== 'Waterfall'
+      ? `${props.report} - ${clearCarets(props.patient.subjectName)}`
+      : '';
+  return (
+    <>
+      {/* <Draggable> */}
```
Comment (OzgeYurtsever) [inline src/components/searchView/Report.jsx:360]: clean code here

File: src/services/annotationServices.js [hunk 9]
```diff
@@ -154,6 +154,19 @@ export function uploadSegmentation(segmentation, segName, projectId = "lite") {
   return http.post(url, segData, config);
 }
 
+export function getReport(projectId, patId, filter) {
```
Comment (OzgeYurtsever) [inline src/services/annotationServices.js:157]: move this to reportServices.js file

**Review by emelalkim — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (OzgeYurtsever) [reply src/App.js:92]: Done

File: src/App.js (hunk 2 above)
Comment (OzgeYurtsever) [reply src/App.js:881]: Done

File: src/components/annotationsList/reducer.js (hunk 3 above)
Comment (OzgeYurtsever) [reply src/components/annotationsList/reducer.js:547]: Done

File: src/components/annotationsList/types.js (hunk 4 above)
Comment (OzgeYurtsever) [reply src/components/annotationsList/types.js:50]: Done

File: src/components/common/SelectModalMenu.jsx (hunk 5 above)
Comment (OzgeYurtsever) [reply src/components/common/SelectModalMenu.jsx:50]: Done

File: src/components/searchView/Report.jsx (hunk 6 above)
Comment (OzgeYurtsever) [reply src/components/searchView/Report.jsx:39]: Done

File: src/components/searchView/Report.jsx (hunk 7 above)
Comment (OzgeYurtsever) [reply src/components/searchView/Report.jsx:132]: Done

File: src/components/searchView/Report.jsx (hunk 8 above)
Comment (OzgeYurtsever) [reply src/components/searchView/Report.jsx:360]: Done

File: src/services/annotationServices.js (hunk 9 above)
Comment (OzgeYurtsever) [reply src/services/annotationServices.js:157]: Done

**Pull request merged**